src/Raven.Client/Http/LoadBalancingRoundRobinNodeSelector.cs

```diff
@@ -0,0 +1,31 @@
+﻿using System.Threading;
+
+namespace Raven.Client.Http
+{
+    public class LoadBalancingRoundRobinNodeSelector : FailoverNodeSelector
+    {
+        private int _syncFlag;
+
+        public LoadBalancingRoundRobinNodeSelector(Topology topology) : base(topology)
+        {
+            _syncFlag = 0;
+        }
+
+        public override void OnSucceededRequest(int nodeIndex)
+        {
+            if (Interlocked.CompareExchange(ref _syncFlag, 1, 0) == 1)
+                return;
+
+            var topology = _topology;
+            if (topology.Nodes.Count == 0)
+                RequestExecutor.ThrowEmptyTopology();
```

Comment:
You are throwing here, and you left the `_syncFlag` on, killing it for all time.